scripts/pre_commit_linter.py

```diff
@@ -2227,6 +2228,40 @@ def _check_codeowner_file(self):
 
         return summary_messages
 
+    def _check_extra_js_files(self):
+        """Checks if the changes made include extra js files in core
+        or extensions folder which are not specified in
+        build.JS_FILEPATHS_NOT_TO_BUILD.
+        """
+        summary_messages = []
+        js_files_to_check = [
+            filepath for filepath in self.all_filepaths if (
+                filepath.endswith('.js'))]
+
+        extra_js_files = []
+        for filename in js_files_to_check:
```

Comment:
@ankita240796, I think we redirect print to stdout (using _redirect_stdout).

Maybe we can have a simple and clean error message around the following pattern:
```
Starting extra js files check
----------------------------------------
a/b/c/d.js --> Found extra .js file
e/f/g/h.js --> Found extra .js file

If you want the above files to be present as js files, add them to the list JS_FILEPATHS_NOT_TO_BUILD in build.py else rename them to .ts
----------------------------------------
FAILED Extra js files check
``` 


(You can follow any existing pattern for checks and see how the messages are printed.)

Reply:
Done.